python/lsst/analysis/tools/tasks/metricAnalysis.py

```diff
@@ -26,33 +26,88 @@
 )
 
 
+from lsst.pex.config import ListField
 from lsst.pipe.base import connectionTypes as ct
 
 from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
 
 
 class MetricAnalysisConnections(
     AnalysisBaseConnections,
-    dimensions=("skymap",),
-    defaultTemplates={"metricBundleName": "objectTableCore_metrics"},
+    dimensions=(),
+    defaultTemplates={"metricTableName": ""},
 ):
+
     data = ct.Input(
-        doc="A summary table of all metrics by tract.",
-        name="{metricBundleName}Table",
+        doc="A table containing metrics.",
+        name="{metricTableName}",
         storageClass="ArrowAstropy",
-        dimensions=("skymap",),
         deferLoad=True,
+        dimensions=(),
     )
 
+    def __init__(self, *, config=None):
+
+        self.dimensions.update(frozenset(sorted(config.outputDataDimensions)))
+        super().__init__(config=config)
+        self.data = ct.Input(
+            doc=self.data.doc,
+            name=self.data.name,
+            storageClass=self.data.storageClass,
+            deferLoad=self.data.deferLoad,
+            dimensions=frozenset(sorted(config.inputDataDimensions)),
+        )
+
 
-class MetricAnalysisConfig(AnalysisBaseConfig, pipelineConnections=MetricAnalysisConnections):
-    pass
+class MetricAnalysisConfig(
+    AnalysisBaseConfig,
+    pipelineConnections=MetricAnalysisConnections,
+):
+    inputDataDimensions = ListField[str](
+        doc="Dimensions of the input data table.",
+        default=(),
+        optional=False,
+    )
+    outputDataDimensions = ListField[str](
+        doc="Dimensions of the outputs.",
+        default=(),
+        optional=False,
+    )
 
 
 class MetricAnalysisTask(AnalysisPipelineTask):
-    """Turn metric bundles which are per tract into a
-    summary metric table.
-    """
+    """Perform an analysis of a metric table."""
 
     ConfigClass = MetricAnalysisConfig
     _DefaultName = "metricAnalysis"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        """Take a metric table and run an analysis tool on the
+        data it contains. This could include creating a plot
+        the metrics and/or calculating summary values of those
+        metrics, such as means, medians, etc. The analysis
+        is outlined within the analysis tool.
```

Comment:
I would just putting this in doc for the class, and replacing this entire docstring with a comment `# Docstring inherited.`